internal/crypto/src/time_stamp/provider.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use async_trait::async_trait;
+use bcder::{encode::Values, OctetString};
+use rand::{thread_rng, Rng};
+use x509_certificate::DigestAlgorithm;
+
+use crate::{asn1::rfc3161::TimeStampReq, time_stamp::TimeStampError};
+
+/// A `TimeStampProvider` implementation can contact a [RFC 3161] time stamp
+/// service and generate a corresponding time stamp for a specific piece of
+/// data.
+///
+/// [RFC 3161]: https://datatracker.ietf.org/doc/html/rfc3161
+pub trait TimeStampProvider {
```

Comment:
Provide a builder interface over a trait.

Reply:
Won't implement in this PR. I don't really agree with the feedback, so not tracking as an issue. If there's a larger pattern change, let's discuss in architecture/API review.

Reply:
Yeah, not sure what a  build would build.  Also, timestamp in in the Signer interface and only used when we are doing the fetching of the response.  Signers can implement their own implementation and return the time stamp response.